### DISCUSSION TITLE
Dynamic landscape error

### DISCUSSION THREAD

**Discussion by ellalewi547** [RangeShifter/RangeshiftR-tutorials#53] [General] [answered]
Hi there I was wondering if you could help me work out what's wrong with my dynamic landscape. I would like to run a three stage landscape where the first 0-10 years is the existing woodland cover, 10-40 years is 10% woodland increase and then 40-100years is 20% woodland increase. 

I have added the three different landscape files and then the patch layer I have kept the same:

`year_blocks <- c(0,10,40
land <- ImportedLandscape(LandscapeFile = c("Final land cover V6.asc", "Restoration Scenario.asc", "Stage 2 Restoration Creation.asc"), 
                          PatchFile = c("Final habitat patch V7.asc", "Final habitat patch V7.asc", "Final habitat patch V7.asc"),
                          Resolution = 25, 
                          DynamicLandYears = year_blocks,
                          Nhabitats = 10,                                      
                          K_or_DensDep = c(0.0047, 0,0,0,0, 0, 0, 0, 0, 0)) 
`

I have not made any other changes to the script from my existing model which works fine.

I have checked each landscape file and they are all match extent, cell size etc

I am getting the following error message:

`> RunRS(master_parameter,dirpath)
Checking Control parameters 

Control Parameters checked

Run Simulation(s) with seed 324135 ...
LandscapeFile headers OK: C:/Rangeshifter/Inputs/Final land cover V6.asc
PatchFile headers OK: C:/Rangeshifter/Inputs/Final habitat patch V7.asc
Dynamic landscape habitat map #1, headers OK: C:/Rangeshifter/Inputs/Restoration Scenario.asc
Dynamic landscape patch map #1, headers OK: C:/Rangeshifter/Inputs/Final habitat patch V7.asc
Dynamic landscape habitat map #2, headers OK: C:/Rangeshifter/Inputs/Stage 2 Restoration Creation.asc
Dynamic landscape patch map #2, headers OK: C:/Rangeshifter/Inputs/Final habitat patch V7.asc

Error reading landscape 0 - aborting

***** Elapsed time: 3716 seconds

*****
***** Simulation completed 
***** Outputs folder: C:/Rangeshifter/Outputs/
*****
$Errors
[1] 0`

Would you be able to tell me where I could be going wrong or would you need more detail to make any inferences?

Many thanks,

Ella

**Comment by ellalewi547** [comment]
Just to add to this, I've also attempted to convert the land cover rasters into matrix format and this is the error I am getting:

eg. 

`> land3 <-  raster(paste0(dirpath,"Inputs/Landscape_3.asc"))`
`> land_matrix3 <- as.matrix(land3, ncol = ncol(land3), nrow = nrow(land3), byrow = TRUE)`

```
land <- ImportedLandscape(LandscapeFile = c(land_matrix1, land_matrix2, land_matrix3),
+                           PatchFile = c("Final habitat patch V7.asc", "Final habitat patch V7.asc", "Final habitat patch V7.asc"),
+                           Resolution = 25, 
+                           DynamicLandYears = year_blocks,
+                           Nhabitats = 10,                                      
+                           K_or_DensDep = c(0.0047, 0,0,0,0, 0, 0, 0, 0, 0)) 
Error in validObject(.Object) : 
  invalid class “ImportedLandscape” object: invalid object for slot "LandscapeFile" in class "ImportedLandscape": got class "numeric", should be or extend class "character"
```
  

 This is my version of the package:
  > library(RangeShiftR)
RangeshiftR version 1.1.1 (05.07.2024)
Copyright (C) 2020-2024 Anne-Kathleen Malchow, Greta Bocedi, Stephen C.F. Palmer, Justin M.J. Travis, Jette Reeg, Damaris Zurell

This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY.
You are welcome to redistribute it and/or modify it under certain conditions;
type 'RangeShiftR_license()' for details.

**Comment by JetteReeg** [comment, marked as answer]
Hi Ella, 
I am sorry to come back to you so late.
To your original post:
Have you tried running each landscape on its own without dynamic landscapes? It is difficult to tackle down the issue without a minimal working example or at least the landscape files. A wild guess would be that the NA values in your landscapes do not match. Have you checked that? (really check if for all landscape, cells with NAs are constant and do not change.

To your reply: 
The error message clearly says that LandscapeFile should be of class "character". You can only provide a list of file names, but not a matrix. (This is only a feature of a side project, which might be added in future RangeShiftR releases.)

**Reply by ellalewi547** [reply]
Oh I see thank you very much! I've just checked the summary of each raster and one of them doesn't match the no. of NA's so that must be it, thanks alot.
